packages/api-explorer/src/components/SideNav/SideNav.tsx

```diff
@@ -113,19 +113,20 @@ export const SideNav: FC<SideNavProps> = ({ headless = false, spec }) => {
     const searchParams = new URLSearchParams(location.search)
     if (debouncedPattern && debouncedPattern !== searchParams.get('s')) {
       searchParams.set('s', debouncedPattern)
-      navigate(location.pathname, { search: searchParams.toString() })
+      navigate(location.pathname, { s: searchParams.get('s') })
     } else if (!debouncedPattern && searchParams.get('s')) {
       searchParams.delete('s')
-      navigate(location.pathname, { search: searchParams.toString() })
+      navigate(location.pathname, { s: null })
     }
-  }, [location.search, debouncedPattern])
+  }, [debouncedPattern])
```

Comment:
@jkaster 
I have to remove `location.search` as a dependency from this useEffect due to routing backwards/forwards. Imagine there you load APIExplorer, leave it blank, then search ‘test’ in the box. After it searches, if you press the back button, APIExplorer will have its listener trigger and set our search pattern to ‘’.  *The problem, however, is if `location.search` is a dependency of this useEffect.  The pattern (of value '') will not have debounced immediately and the `debouncedPattern` will refer to ‘test’. Hence the URL will not remove the 's=' parameter as it should

Reply:
I think that may have been feedback from @bryans99 instead of me? If it was me, sorry I don't remember!

Reply:
Totally correct, my apologies ! This is for Bryn.

Reply:
If its causing a problem with it and not without then lets remove it. Please add a comment as to why its not included. At some point I will look at it more closely